src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/WarningPropertiesCollection.cs

```diff
@@ -160,6 +160,7 @@ public static void ApplyProjectWideWarningsAsErrorProperties(ILogMessage message
                 {
                     // If the project wide AllWarningsAsErrors is true and the message has a valid code or
                     // Project wide WarningsAsErrors contains the message code then upgrade to error.
+                    message.Message = Strings.WarningAsError + ": " + message.Message;
```

Comment:
I would prefer using a special string resource to concatenate the result. That way, message in different languages can adjust the semicolon if needed.

```suggestion
                    message.Message = string.Format(CultureInfo.CurrentCulture,  Strings.WarningAsError, message.Message);
```